### ISSUE TITLE
Navigation indexes are not rendered correctly on the Blog index from the second page onwards

### ISSUE THREAD

**Issue by vedranmiletic** [squidfunk/mkdocs-material#5972]
### Bug description

When using `blog` plugin with `navigation.indexes` feature, only the first page of the Blog index has the correctly rendered navigation. Other pages of the Blog index are rendered as if the feature is not enabled.

Interestingly enough, this happens only when there is at least three pages of posts in the Blog index.

Blog archive is affected the same way. I haven't checked if categories are affected since we don't use them.

Additionally, if the `tags` plugin is enabled, the Tags page will exhibit the same behavior regarding the rendering of the Blog index in the navigation. I have not included this in the minimal example on purpose to keep it minimal, but it can be observed on [our group website](https://group.miletic.net/en/blog/tags/). I can prepare a less minimal example if needed. 

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Setting up navigation](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/)

### Reproduction

[9.2.8-navigation.zip](https://github.com/squidfunk/mkdocs-material/files/12526742/9.2.8-navigation.zip)

### Steps to reproduce

1. `mkdocs serve`
2. look at the navigation in first page of the Blog index
3. see it is rendered as intended
4. navigate to the second or the third page
5. see it is rendered as if it would be without `navigation.indexes` feature enabled

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by balansczerni** [comment]
I guess I have the same error? 

I thought it was because I set `tags_extra_files`. In my case, the error occurs when more than 10 files appear in the `posts` directory and only occurs with (and from location of) `tags_extra_files`.

It's hard for me to describe it better so I'm attaching a minimal reproduction:
[9.2.8+insiders.4.40.4-blog-vs-indexes.zip](https://github.com/squidfunk/mkdocs-material/files/12551862/9.2.8%2Binsiders.4.40.4-blog-vs-indexes.zip)

```
.
├─ docs/
│  └─ blog/
│     ├─ subset.md
│     ├─ posts/
│     └─ index.md
└─ mkdocs.yml
```

mkdocs.yml:

```YAML
theme:
  name: material
  features:
    - navigation.indexes

plugins:
  #- info
  - blog
  - tags:
      tags_extra_files:
        blog/subset.md:
          - mytag

nav:
  - Page: index.md
  - Blog:
    - blog/index.md
    - Subset page: blog/subset.md  
  - Tags: tags.md

extra:
  tags:
    My tag: mytag
```

![Zrzut ekranu 2023-09-7 o 19 14 24 Średni](https://github.com/squidfunk/mkdocs-material/assets/123209821/45d06e34-0b53-42ab-8603-44f3a389afbb)

![Zrzut ekranu 2023-09-7 o 19 14 49 Średni](https://github.com/squidfunk/mkdocs-material/assets/123209821/69490394-fa0d-4b15-993c-fb638583d690)

In addition, I think I have located the place in the code where the bug takes place (this is from `nav-item.html` with my comments):

```
{% macro render(nav_item, path, level) %}
  {% set class = "md-nav__item" %}
  {% if nav_item.active %}
    {% set class = class ~ " md-nav__item--active" %}
  {% endif %}
  {% if nav_item.children %}
    {% set indexes = [] %}
    {% if "navigation.indexes" in features %}
      {% for nav_item in nav_item.children %}
        {% if nav_item.is_index and not index is defined %}
          {% set _ = indexes.append(nav_item) %}
        {% endif %}
      {% endfor %}
      {% endif %}
  
    {% if "navigation.sections" in features and level == 1 + (
      "navigation.tabs" in features
    ) %}
      {% set class = class ~ " md-nav__item--section" %}
    {% elif not nav_item.active and "navigation.prune" in features %}
      {% set class = class ~ " md-nav__item--pruned" %}
      {% set prune = true %}
    {% endif %}
    <li class="{{ class }} md-nav__item--nested">
      {% if not prune %}
        {% set expanded = "navigation.expand" in features %}
        {% set active = nav_item.active or expanded %}
        {% set checked = "checked" if nav_item.active %}
        {% if expanded and not checked %}
          {% set indeterminate = "md-toggle--indeterminate" %}
        {% endif %}
        <input class="md-nav__toggle md-toggle {{ indeterminate }}" type="checkbox" id="{{ path }}" {{ checked }}>
        {% if not indexes %}
          <label class="md-nav__link" for="{{ path }}" id="{{ path }}_label" tabindex="0">
            {{ render_content(nav_item) }} 🆘 This is "extra" blog link.
            <span class="md-nav__icon md-icon"></span>
          </label>
        {% else %}
          {% set index = indexes | first %}
          {% set class = "md-nav__link--active" if index == page %}
          <div class="md-nav__link md-nav__container">
            <a href="{{ index.url | url }}" class="md-nav__link {{ class }}">
              {{ render_content(index, nav_item) }} ✅ This is proper link for blog.
            </a>
            {% if nav_item.children | length > 1 %}
              <label class="md-nav__link {{ class }}" for="{{ path }}">
                <span class="md-nav__icon md-icon"></span>
              </label>
            {% endif %}
          </div>
        {% endif %}
```

**Comment by squidfunk** [comment]
Thanks to both of you for the reproductions – we'll look into it asap!

**Comment by squidfunk** [comment]
Fixed in 433c137bc. Paginated views did not correctly check if the base view is an index page or not. Now, if the view is an index page (i.e. is called `index.md` or `README.md`), paginated pages are also generated as index pages.

@smokedlinq I'm not sure where or how this happens for non-blog sites, as the issue reported is exclusive to the blog plugin. Please open a new issue with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/), so we can look into it.

**Comment by squidfunk** [comment]
@vedranmiletic if this does not solve the problem for the tags plugin, please create a new issue.

**Comment by vedranmiletic** [comment]
@squidfunk thanks, it seems to work correctly now, both for tags page and blog index pages after the first one.

**Comment by squidfunk** [comment]
Released as part of [9.4.2](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.2).
